kork/kork-github/src/test/java/com/netflix/spinnaker/kork/github/GitHubAppAuthenticatorTest.java

```diff
@@ -0,0 +1,224 @@
+          + "kD1jlyzojU8JI6GUPpasMU28SP/fR9NEhtWke7woCiy8oU/ZqlP3pxRx7jzPzG2b\n"
+          + "mJlS5vmfQwKBgQCnAmVxaG9wqZnn7cuLAM5piaaAld/r7zXXDgS7bMa1DIJd9+NP\n"
+          + "vy1pbfpIp65GpRWPCaMznpbBPyDbubHiz5mAOVOwkMo1ZRFXJBACoaNY/wCoCa5Z\n"
+          + "Ct1J694mkZ3PhrWa/8uMpIcDW4SgeZHgFOXY32+a29wFgILr61Emf54K7wKBgC+B\n"
+          + "UNhgWssQaNKeyRcAlTTkf9P/N7lAoOPKYzNhUQDFIbPRTH2dyEwWvHUSDnRIb6Cu\n"
+          + "ujG64/szINOTfnLon2eWXmiZmeRJ4L7NCoCIDF98LKHyqGupTlTrX1CWSzxqem4I\n"
+          + "RM2zLAcXzUPyBSKQSskhFvMTi8UY3UsPwwmvoOqVAoGAIo6R8hkO0mGnEtAtuIp0\n"
+          + "4ZrsJvUQ0F/Nse/IvdT8PBC2xcz6V3f+LaJG9yqN57fGoD6V5tme0hdPb6K//42l\n"
+          + "oBJQ5zTTHMt+RtkKDyPt+K3reXIxFL4E0GiYrJpQRsjKu5xWZY73jO4zC/9aTP9y\n"
+          + "0LJp16baAExMUyvsgf30c4A=\n"
+          + "-----END PRIVATE KEY-----";
+
+  private Path tempPrivateKeyFile;
+
+  @BeforeEach
+  void setUp(@TempDir Path tempDir) throws Exception {
+    // Write pre-generated test private key to temp file
+    tempPrivateKeyFile = tempDir.resolve("test-private-key.pem");
+    Files.write(tempPrivateKeyFile, TEST_PRIVATE_KEY.getBytes());
+  }
+
+  @Test
+  void shouldCreateAuthenticatorWithValidConfiguration() {
+    // When & Then
+    assertDoesNotThrow(
+        () ->
+            new GitHubAppAuthenticator(
+                "12345", tempPrivateKeyFile.toString(), "67890", "https://api.github.com"));
+  }
+
+  @Test
+  void shouldFailWithInvalidPrivateKeyPath() {
+    // When & Then
+    RuntimeException exception =
+        assertThrows(
+            RuntimeException.class,
+            () ->
+                new GitHubAppAuthenticator(
+                    "12345", "/nonexistent/path/key.pem", "67890", "https://api.github.com"));
+
+    assertTrue(exception.getMessage().contains("Failed to load GitHub App private key"));
+  }
+
+  @Test
+  void shouldNormalizeBaseUrlWithTrailingSlash() {
+    // Given
+    GitHubAppAuthenticator authenticator =
+        new GitHubAppAuthenticator(
+            "12345", tempPrivateKeyFile.toString(), "67890", "https://api.github.com/");
+
+    // When - internal baseUrl should have trailing slash removed
+    // We can't test this directly without exposing internal state,
+    // but we can verify the authenticator was created successfully
+    assertNotNull(authenticator);
+  }
+
+  @Test
+  void shouldSupportPKCS8Format() throws IOException {
+    // Given - TEST_PRIVATE_KEY is in PKCS#8 format (BEGIN PRIVATE KEY)
+    GitHubAppAuthenticator authenticator =
+        new GitHubAppAuthenticator(
+            "12345", tempPrivateKeyFile.toString(), "67890", "https://api.github.com");
+
+    // When & Then - should load successfully
+    assertNotNull(authenticator);
+  }
+
+  @Test
+  void shouldSupportPKCS1Format(@TempDir Path tempDir) throws IOException {
+    // Given - PKCS#1 format (BEGIN RSA PRIVATE KEY)
+    String pkcs1Key =
+        "-----BEGIN RSA PRIVATE KEY-----\n"
```

Comment:
NIT:  And definitely not a big deal but should be able to use multiline strings with java 17 vs. the newline and quote sorta thing here.

Reply:
Can fix it. But currently waiting for our security team to approve this, as they have to make an exception to allow commiting RSA private keys, one more time.